### PR TITLE
Load tun kernel module

### PR DESCRIPTION
**Pull request by SchSeba**
The sriov-network-config-daemon will try to load the tun kernel driver when started

This PR is needed to continue to the PR merged in the sriov-device-plugin to allow the mount of tun device inside the container to create TAP devices.

The TAP devices are in use by dpdk applications that want to interact using a slow path with the Linux kernel networking.

### CODE REVIEW

File: config/crd/bases/sriovnetwork.openshift.io_sriovnetworknodestates.yaml [hunk 1]
```diff
@@ -50,6 +50,8 @@ spec:
                       type: integer
                     name:
                       type: string
+                    needVhostNet:
```
Comment (zshi-redhat) [inline config/crd/bases/sriovnetwork.openshift.io_sriovnetworknodestates.yaml:53]: I prefer to not changing API and always loading the tun module like how rdma_core is installed, do you see any issue loading the tun module unconditionally?

Comment (SchSeba) [reply config/crd/bases/sriovnetwork.openshift.io_sriovnetworknodestates.yaml:53]: sounds good to me :)

**Comment by SchSeba** [comment]
@zshi-redhat done can you have another round of review?

BTW do you think we should do the same for the vfio-pci driver? just load it in the daemon initialization?

**Comment by zshi-redhat** [comment]
> @zshi-redhat done can you have another round of review?
> 
> BTW do you think we should do the same for the vfio-pci driver? just load it in the daemon initialization?

It doesn't hurt to remove, but I think the current loading of vfio-pci module has more precise per-device control than tun or rdma modules. The main reason of not adding the same per-device control for tun module is that we don't want to change API in the backport or just for the module load. so I think we can leave the vfio-pci as it is.

**Review by zshi-redhat — APPROVED** [review]

**Comment by zshi-redhat** [comment]
> Expose the Vhost field to the sriov-network-config-daemon, so it will be able to load the tun kernel driver when needed
> 
nit: this commit message need change to reflect current implemention.

> This PR is needed to continue to the PR merged in the sriov-device-plugin to allow the mount of tun device inside the container to create TAP devices.
> 
> The TAP devices are in use by dpdk applications that want to interact using a slow path with the Linux kernel networking.

**Comment by SchSeba** [comment]
Hi @zshi-redhat I fixed the commits

**Review by zshi-redhat — APPROVED** [review]

File: test/conformance/tests/sriov_operator.go [hunk 2]
```diff
@@ -2004,6 +2022,17 @@ func createCustomTestPod(node string, networks []string, hostNetwork bool) *k8sv
 			node,
 		)
 	}
+
+	if podCapabilies != nil && len(podCapabilies) != 0 {
```
Comment (Eoghan1232) [inline test/conformance/tests/sriov_operator.go:2026]: Should it be podCapabilies -> podCapabilities ?

File: test/conformance/tests/sriov_operator.go [hunk 3]
```diff
@@ -1991,10 +2009,10 @@ func isPodConditionUnschedulable(pod *k8sv1.Pod, resourceName string) bool {
 }
 
 func createTestPod(node string, networks []string) *k8sv1.Pod {
-	return createCustomTestPod(node, networks, false)
+	return createCustomTestPod(node, networks, false, nil)
 }
 
-func createCustomTestPod(node string, networks []string, hostNetwork bool) *k8sv1.Pod {
+func createCustomTestPod(node string, networks []string, hostNetwork bool, podCapabilies []corev1.Capability) *k8sv1.Pod {
```
Comment (Eoghan1232) [inline test/conformance/tests/sriov_operator.go:2015]: podCapabilies -> podCapabilities ?

**Comment by SchSeba** [comment]
@Eoghan1232 thanks I miss that!

**Review by Eoghan1232 — APPROVED** [review]
LGTM
Thanks for working on this!

**Comment by SchSeba** [comment]
@pliurh can we merge this PR?

**Review by e0ne — APPROVED** [review]

**Pull request merged**